magento/magento2
# Can't able to delete/rename sitemap.xml in file system

### Description (*)
When we try to delete the generated sitemap, it is deleted from the admin side but not deleted from the file system. So, I have checked, and there is a Delete controller for deleting the sitemap from the admin side. Here is the file - 
> _**/vendor/magento/module-sitemap/Controller/Adminhtml/Sitemap/Delete.php**_

in the **execute()** function, the **`$directory`** is using the ROOT (base) directory instead of the PUB directory at line 54. So, it is checking the file in the root instead of the pub directory at line 72 as in the below image.

![delete-controller-execute-method](https://user-images.githubusercontent.com/59201168/196389956-8236156f-6bd0-4063-be31-5042d24bbeac.png)

The sitemap file is generated under the pub directory. So, it can't able to delete the sitemap file from the pub directory after deleting it from the admin side.

Also, if we rename the sitemap file and Save and Generate it. Then it will generate a new file and it will keep the old one also in the file system. There is a Save controller for saving the sitemap and here is the file -
> _**/vendor/magento/module-sitemap/Controller/Adminhtml/Sitemap/Save.php**_

in the **clearSiteMap()** function, doing the same thing as in the Delete controller like the **`$directory`** is using the ROOT (base) directory instead of the PUB directory at line 131. And that's why when the sitemap is renamed, it can't delete from the file system as in the below image.

![save-controller-clearSiteMap-method](https://user-images.githubusercontent.com/59201168/196390364-4bc1417d-7665-4340-ad1b-67f520e1ad18.png)

So, We need to use the PUB directory at both places.

### Related Pull Requests
None

### Fixed Issues (if relevant)
None

### Manual testing scenarios (*)

1. Go to Marketing -> Sitemap and create a sitemap.
2. Add a Sitemap by clicking on the 'Add Sitemap' button. And fill up the field like the attached image and save the sitemap.

![New-Site-Map-Site-Admin](https://user-images.githubusercontent.com/59201168/196391552-8af240af-d89e-4c99-8ad1-19eb75bed32d.png)

3. Now generate a sitemap from the grid by clicking on the generate link.
4. Sitemap is generated in the pub directory. Now open that sitemap on the admin side and delete it.
5. Check in the pub directory. The sitemap is still there.

### Questions or comments


### Contribution checklist (*)


### Resolved issues:
1. [x] resolves magento/magento2#37468: Can't able to delete/rename sitemap.xml in file system
